### DISCUSSION TITLE
[fixed] No USB (No Wifi, No Keyboard, No USB-Disk) on rpi A+ and rpi zero 1.3, and zero W

### DISCUSSION THREAD

**Discussion by m-sja** [rern/rAudio#860] [Fixed Issues]
Hi, i like the project and it would be a perfect fit for some of my old gen1 PIs.
Sadly i could not get Wifi to work (or USB at all).

Hardware: Raspberry Pi A+ and Raspberry Pi Zero 1.3
Wifi-Dongle: rtl8188 (works with official OSs since ... ever? and volumio)
Image: rAudio-1-RPi0-1-20220214.img.xz

I downloaded the image and burned it to 32G Samsung uSD using pi-imager.
Then renamed "boot/wifi0" to "wifi" and updated SSID and password, saved the file using notepad++ and ejected the card safely.
Boot went fine, but it hangs at around 40s uptime waiting for 3 processes (after "Started User Login Management")
1/3: ... Network Name Resolution
2/3: ... /sys/subsystem/net/devices/wlan0
3/3: ... Load/Save Random Seed
After around 100s it continues to "rAudio login:" followed by some occasionally messages

Wifi-LED on the dongle stays dark. No connection to my wifi, also no AP visible.

If i switch the Dongle for USB Keyboard, the 3 LEDs on the keyboard flash shortly, but then the keyboard does not work, also no Numlock-toggle-led.

Keyboard does also not work when booting with it already attached.

Same on Raspberry A+ as well as on Zero 1.3. (using different newly flashed uSD cards)

I tried with rPi Zero W, and the internal wifi works fine, but USB-Keyboard and USB-Stick does not.
Seems like there are some USB Bus issues.

Any suggestions what to do?

**Comment by R101G** [comment]
Arch-Linux support for wifi dongles is very poor. You will probably have to manually install a driver.

**Reply by m-sja** [reply]
Installing a proper driver should not be the problem, but without ssh (cause no wifi) and without keyboard (cause no usb) the only option would be the serial console via gpio header.

But the main issue is that the image offered for rpi gen1 is missing basic thing like working usb-keyboard support.
A working keyboard on a computer should be the most important thing.
This must be a bug, otherwise offering a image is pointless.

**Reply by rern** [reply]
Just uploaded a new image file, [rAudio-1-RPi0-1-202203271.img.xz](https://github.com/rern/rAudio-1/releases/tag/i20220317), that fixes the USB issue.

Please try it.

**Comment by m-sja** [comment]
Thanks @rern ! This fixed the my problem.
I can confirm:
- USB is working now
- Keyboard working also
- USB Wifi rtl8188 Dongle works out of the box.

**Reply by withakw** [reply]
Hi @rern. 
I discovered rAudio yesterday and I am loving it. Please could you help me?
I am using the latest build 
![image](https://user-images.githubusercontent.com/22167274/219204982-d1c90072-5d4f-4324-bbbd-48fe1d3b899e.png)
and I have tried a couple of USB keyboards, 1 wired and 1 wireless, both with multimedia buttons and neither of them work. Is there something I need to activate to make it work?
Thank you very much for your time
Keith

**Reply by rern** [reply]
Hi Keith,

Features > Browser on RPi - Enable

**Comment by withakw** [comment]
Hi @rern 
Thank you so much for the speedy response.
I tried that but the option turns itself off after a few seconds. The information button about Chromium says it is dead as I have this 
![image](https://user-images.githubusercontent.com/22167274/219313385-332a9673-881a-4387-a437-59b49b7d7c87.png)
